src/utils.ts

```diff
@@ -113,37 +113,36 @@ export function generate(routes: Route[]): string {
   return `\
 // THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.'
 // Run \`yarn nextjs-routes\` to regenerate this file.
+/* eslint-disable */
 
 type Route =
   | ${routes
     .map((route) => {
       const [query, requiredKeys] = getQueryInterface(route.query);
       if (requiredKeys > 0) {
-        return `{ pathname: '${route.pathname}'; query: Query<${query}> }`;
+        return `{ pathname?: '${route.pathname}'; query: Query<${query}> }`;
```

Comment:
@tatethurston The new optional `pathname` here really throws off our types because we were using the `Route` type and expected `pathname` to be defined. This is why I went with the separate `RouteOrQuery` type in my PR and left `Route` untouched.

Reply:
Sorry about that, I can publish an update this afternoon. How are you consuming the Route type? Are you deriving it? And if so, should we find a way to import it directly?

Reply:
The `Route` type is a global ambient type so we're able to "just use it." Here's an example:
```ts
interface NavLinkProps {
  text: string
  href: Route
}

const NavLink = (props: LinkProps) => {
  // this errors now because `pathname` is undefined
  const isIndex = props.href.pathname === '/'

  return (
    <Link href={props.href}>
      <a className={isIndex ? 'bg-gray-500' : 'bg-gray-300'}>{props.text}</a>
    </Link>
  )
```

Reply:
🤦 Right, that is an oversight on my end. I don't like that the global namespace is polluted right now.

I propose the following:
1. Move the ambient type definitions (Route, Query, Pathname) into a modue declaration ('nextjs-routes')
2. export Route from nextjs-routes, so it can be imported: `import type { Route } from 'nextjs-routes'`

Reply:
Sounds good!